rdrf/registry/patients/parent_view.py

```diff
@@ -57,46 +57,46 @@ class ParentView(BaseParentView):
 
     def get(self, request, registry_code):
         context = {}
-        if request.user.is_authenticated:
-            parent = ParentGuardian.objects.get(user=request.user)
-            registry = Registry.objects.get(code=registry_code)
-
-            self.registry = registry
-            self.rdrf_context_manager = RDRFContextManager(self.registry)
-
-            patients_objects = parent.patient.all()
-            patients = []
-
-            forms_objects = RegistryForm.objects.filter(registry=registry).exclude(is_questionnaire=True).order_by('position')
-
-            progress = form_progress.FormProgress(registry)
-
-            for patient in patients_objects:
-                forms = []
-                for form in forms_objects:
-                    if form.is_questionnaire or not request.user.can_view(form):
-                        continue
-                    forms.append({
-                        "form": form,
-                        "progress": progress.get_form_progress(form, patient),
-                        "current": progress.get_form_currency(form, patient),
-                        "readonly": request.user.has_perm("rdrf.form_%s_is_readonly" % form.id)
-                    })
-
-                rdrf_context = self.rdrf_context_manager.get_or_create_default_context(patient)
-                patients.append({
-                    "patient": patient,
-                    "consent": consent_status_for_patient(registry_code, patient),
-                    "context_id": rdrf_context.pk,
-                    "forms": forms
+
+        parent = ParentGuardian.objects.get(user=request.user)
+        registry = Registry.objects.get(code=registry_code)
+
```

Comment:
- Why do we allow non parents open up the parents page at all?
- We aren't checking that the parent is part of the registry.

Seems too lax to me. 

Also true for creating patients (ie. a parent could create a patient in another registry by just passing in the other registry code).

And what about the `ParentEditView`, I see no checks on the `get`. Anyone logged in can get details of parents?

Anyone allowed to get

Reply:
Updated.